src/Sarif/Core/Run.cs

```diff
@@ -217,5 +219,41 @@ public bool ShouldSerializeColumnKind()
         public bool ShouldSerializeLogicalLocations() { return this.LogicalLocations.HasAtLeastOneNonDefaultValue(LogicalLocation.ValueComparer); }
 
         public bool ShouldSerializeNewlineSequences() { return this.NewlineSequences.HasAtLeastOneNonNullValue(); }
+
+        internal static void ComputePolicies(IEnumerable<ToolComponent> policies)
```

Comment:
>ComputePolicies [](start = 29, length = 15)

Make this API return a dictionary (the cache), then the caller assigns it to PoliciesCache